NEWS.md

```diff
@@ -415,6 +415,8 @@ Library improvements
     * `mktemp` and `mktempdir` now take an optional argument to set which
       directory the temporary file or directory is created in.
 
+    * New, generational garbage collector ([#5227]). In some use cases it can reduce the worst case time for garbage collection by a factor of ten. Julia can now be used for soft real-time applications like audio processing or high-level robot control.
```

Comment:
although I like the enthusiasm, it does not make much sense to speak of absolute numbers (10x) since it is dependent on the size of the heap. It is also sensitive to the workload and the gc improvements can definitely make everything slower in some cases : the test code you used is very friendly to how it works now.

I'm also not sure about the "now usable for soft-rt" part, since old gen collections are still bad latency wise. You will hit some of those regularly on an application where some memory survives longer than in your example.